File: backend/src/main/java/com/app/backend/dtos/BoCauHoiDTO.java
package com.app.backend.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class BoCauHoiDTO {

    @NotBlank
    @JsonProperty("tieu_de")
    private String tieuDe;

    @JsonProperty("mo_ta")
    private String moTa;

    @NotNull
    @JsonProperty("chu_de_id")
    private Long chuDeId; // FK -> ChuDe

    @NotBlank
    @JsonProperty("che_do_hien_thi")
    private String cheDoHienThi; // PUBLIC | PRIVATE
}

File: frontend/src/app/dtos/bo-cau-hoi/bocauhoi-dto.ts
import {IsString} from 'class-validator';

export class BoCauHoiDTO {
  @IsString()
  tieu_de: string;

  @IsString()
  mo_ta?: string;

  chu_de_id: number;

  @IsString()
  che_do_hien_thi: string;// "PUBLIC" | "PRIVATE"

  constructor(data: Partial<BoCauHoiDTO>) {
    this.tieu_de = data.tieu_de ?? '';
    this.mo_ta = data.mo_ta ?? '';
    this.chu_de_id = data.chu_de_id ?? 0;
    this.che_do_hien_thi = data.che_do_hien_thi ?? 'PUBLIC';
  }
}
